### ISSUE TITLE
Scanner crashes if build method of StatefulWidget is being called while scanning is active

### ISSUE THREAD

**Issue by peukerjonathan** [juliansteenbakker/mobile_scanner#404]
Tested with version [v3.0.0-beta.2](https://github.com/juliansteenbakker/mobile_scanner/releases/tag/v3.0.0-beta.2)

If some code in a `StatefulWidget` causes the `build` method of the Widget to execute while the scanner is open the scanner goes blank and gives a `PlatformException(MobileScanner, Called stop() while already stopped!`

This seems to be the case if the `MobileScannerController` constructor is being called inside the `build` method.
However, while moving the `MobileScannerController`  constructor out of the `build` method is avoiding the crashing this leads to another situation: In this case the scanner is only recognizing one barcode and then ignores all further codes (`onDetect` is not triggered anymore, even if the scanner is open)

The only way I found to avoid this is carefuly keeping track of the scanner being active or not - and only call `setState()` when the scanner is not active. Currently I am using this as a "workaround" in my app.

This behaviour can be reproduced on Android and iOS.
I am not sure if this is intended behaviour or if there is an underlying problem here. 

I have created a litte test app which demonstrates all three scenarios (Scanner crashing, Scanner only scanning once, and "Working").
[scanner_problem_test_archive.zip](https://github.com/juliansteenbakker/mobile_scanner/files/10186645/scanner_problem_test_archive.zip)

**Comment by juliansteenbakker** [comment]
Can you please try this with version 3.0.0?

**Comment by peukerjonathan** [comment]
@juliansteenbakker I just used my test app attached to the initial post and upgraded the dependencies to use 3.0.0. Here are my findings:

Case 1: 
**Initial scenario:** Scanner was previously crashing with  `PlatformException(MobileScanner, Called stop() while already stopped!` when `build` method was called by `setState` while the scanner was open.
**With 3.0.0.:** Seems to be fixed, everything is working as expected.

Case 2:
**Initial scenario:** `MobileScannerController` constructor was moved outside of the `build` method -> Scanner was not crashing but did not perform any scans after the first one.
**With 3.0.0.:** No changes - problem persists.

Case 3:
**Initial scenario:** Code modified to only call `setState` when scanner ist not active. This code was already working, just felt "hacky".
**With 3.0.0.:** Still working ;)

So from my point of view the problem is solved when the code is implemented as in case 1 which perfectly aligns with the examples given in this repo.
The only remaining question is, wether the problem in case 2 is an intended behaviour or not.
